docs/user-guide/finalization_stages_timeouts.md

```diff
@@ -0,0 +1,22 @@
+# Finalization Stages & Timeouts
+
+## Overview
+The finalization process in Assisted Service consists of multiple stages, each with a predefined timeout value. These timeouts determine how long the system waits for each stage before considering it as failed.
```

Comment:
Lets add few words about the fact that this is the last stage before moving to `installed` in a successful installation, and that these values are configurable via env vars